js/web/components/Profile.js

```diff
@@ -0,0 +1,330 @@
+
+const Dots = dynamic(() => import('./Dots'))
+const ProfileReleaseTable = dynamic(() => import('./ProfileReleaseTable'))
+const ProfileHubs = dynamic(() => import('./ProfileHubs'))
+const ProfileToggle = dynamic(() => import('./ProfileToggle'))
+
+const Profile = ({ userId }) => {
+  
+  const {
+    getUserCollectionAndPublished,
+    releaseState,
+
+    filterReleasesPublishedByUser,
+
+    filterReleasesList,
+  } = useContext(Release.Context)
+
+  const { getHubsForUser, filterHubsForUser, hubState } = useContext(
+    Hub.Context
+  )
+
+  const { resetQueueWithPlaylist } = useContext(Audio.Context)
+
+  const [profilePublishedReleases, setProfilePublishedReleases] = useState([])
+  const [profileCollectionReleases, setProfileCollectionReleases] = useState()
+  const [profileHubs, setProfileHubs] = useState([])
+  const [view, setView] = useState('releases')
+  const { enqueueSnackbar, closeSnackbar } = useSnackbar()
+  const [profileCollectionIds, setProfileCollectionIds] = useState(undefined)
+  const [toggleView, setToggleView] = useState('releases')
+  const [fetchedUser, setFetchedUser] = useState(false)
+  const [fetchedReleases, setFetchedReleases] = useState(false)
+  const [fetchedHubs, setFetchedHubs] = useState(false)
+  const [fetchedCollection, setFetchedCollection] = useState(false)
+  // const [profileArtistData, setProfileArtistData] = useState([])
+
+  const artistNames = useMemo(
+    () => {
+      if (profilePublishedReleases.length > 0) {
+          return [
+          ...new Set(
+            profilePublishedReleases.map(
+              (release) => release.metadata.properties.artist
+            )
+          ),
+        ] 
+      }
+  },
+    [profilePublishedReleases]
+  )
+
+  useEffect(() => {
+    setFetchedUser(false)
+    const getUserData = async (userId) => {
+      await getHubsForUser(userId)
+      const [collectionIds] = await getUserCollectionAndPublished(userId)
+      setProfileCollectionIds(collectionIds)
+    }
+    if (userId) {
+      setFetchedUser(true)
+      getUserData(userId)
+    }
+  }, [userId])
+
+  useEffect(() => {
+    setFetchedReleases(false)
+    setFetchedCollection(false)
+    if (profileCollectionIds?.length > 0 && userId) {
+      setProfileCollectionReleases(filterReleasesList(profileCollectionIds))
+      const releases = filterReleasesPublishedByUser(userId)
+      setProfilePublishedReleases(releases)
+      setFetchedReleases(true)
+      setFetchedCollection(true)
+    }
+  }, [releaseState, profileCollectionIds])
+
+  useEffect(() => {
+    setFetchedHubs(false)
+    const hubs = filterHubsForUser(userId)
+    if (hubs) {
+      setProfileHubs(hubs)
+      setFetchedHubs(true)
+    }
+  }, [hubState])
+
+
+  const releasesClickHandler = () => {
+    setView('releases')
+    setToggleView('releases')
+  }
+  const hubsClickHandler = () => {
+    setView('hubs')
+    setToggleView('hubs')
+  }
+  const collectionClickHandler = () => {
+    setView('collection')
```

Comment:
can hubs, releases, collection be set up as an array that uses the index? this will keep us flexible if we add things to profile